File: apps/web/src/types/ai-signal.ts
export type AiSignalClass =
  | 'STRONG_BUY'
  | 'BUY'
  | 'NEUTRAL'
  | 'SELL'
  | 'STRONG_SELL';

export type ImpactDirection = 'POSITIVE' | 'NEGATIVE' | 'NEUTRAL';

export interface IndicatorInterpretation {
  indicator: string;
  value: string;
  meaningKo: string;
}

export interface NewsImpact {
  titleKo: string;
  impact: ImpactDirection;
  reasonKo: string;
  hoursAgo?: number | null;
}

export interface SignalPerspective {
  signal: AiSignalClass;
  confidence: number;
  rationale: string[];
  risks: string[];
  summaryKo: string;
  beginnerExplanation?: string | null;
  indicatorInterpretation?: IndicatorInterpretation[] | null;
  newsImpact?: NewsImpact[] | null;
  whatToWatch?: string[] | null;
}

export type TimingVerdictType = 'NOW' | 'NOT_YET' | 'UNCERTAIN';

export interface TimingFactor {
  factor: string;
  detail: string;
  weight: number;
}

export interface TimingVerdict {
  verdict: TimingVerdictType;
  score: number;
  factorsMet: TimingFactor[];
  factorsUnmet: TimingFactor[];
  summaryKo: string;
  disclaimerKo: string;
}

export interface AiSignal {
  ticker: string;
  shortTerm: SignalPerspective;
  longTerm: SignalPerspective;
  timing?: TimingVerdict | null;
  generatedAt: string;
  modelName: string;
  disclaimer: string;
  fallback: boolean;
}

File: apps/api/src/main/java/com/nowini/ai/domain/AiSignal.java
package com.nowini.ai.domain;

import java.time.Instant;

public record AiSignal(
        String ticker,
        SignalPerspective shortTerm,
        SignalPerspective longTerm,
        Instant generatedAt,
        String modelName,
        String disclaimer,
        boolean fallback
) {
    // v2 하위 호환: audit 엔티티가 기존 레코드에 사용
    public enum Signal { STRONG_BUY, BUY, NEUTRAL, SELL, STRONG_SELL }
    public enum Timeframe { SHORT, MID, LONG }
}
